frame/support/procedural/src/lib.rs

```diff
@@ -432,3 +433,37 @@ pub fn crate_to_pallet_version(input: TokenStream) -> TokenStream {
 /// The number of module instances supported by the runtime, starting at index 1,
 /// and up to `NUMBER_OF_INSTANCE`.
 pub(crate) const NUMBER_OF_INSTANCE: u8 = 16;
+
+/// Derive [`MaxEncodedLen`][frame_support::traits::MaxEncodedLen].
+///
+/// # Examples
+///
+/// ```ignore
+/// # // This example is ignored because testing it properly requires imports which would be
+/// # // circular in this context.
+/// # use codec::Encode;
+/// # use frame_support::MaxEncodedLen;
+/// #[derive(Encode, MaxEncodedLen)]
+/// struct TupleStruct(u8, u32);
+///
+/// assert_eq!(TupleStruct::max_encoded_len(), u8::max_encoded_len() + u32::max_encoded_len());
+/// ```
+///
+/// ```ignore
+/// # // This example is ignored because testing it properly requires imports which would be
```

Comment:
As with all other macros. Please move the docs to the re-export in `frame-support`. There you can make this example compile.

Ingnored examples are rather useless as they can break relative fast.

Reply:
Ok: https://github.com/paritytech/substrate/pull/8737/commits/f15e84728d30247099cf9898b16dc0f9617984ce.